src/ucp/am/eager_single.c

```diff
@@ -115,6 +113,8 @@ ucp_am_eager_short_proto_init_common(const ucp_proto_init_params_t *init_params,
         .tl_cap_flags        = UCT_IFACE_FLAG_AM_SHORT
     };
 
+    ucp_proto_short_memtype_params(&params.super, select_param->mem_type);
```

Comment:
maybe we can move this to proto common logic?
i.e keep `super.memtype_op = UCT_EP_OP_LAST` definition, and the common logic that checks memtype_op value will set max_length to 0 if select_param->mem_type is not HOST (instead of returning UNSUPPORTED)